### PR TITLE
Add yamllint pre-commit hook

### LINKED ISSUE #1168: Standardize YAML format
yeah, that's why there should be a shared config so that every PR author wouldn't reformat by accident. I usually have yamllint wired via the pre-commit.com tool

_Originally posted by @webknjaz in https://github.com/aio-libs/aiobotocore/pull/1163#discussion_r1725194345_

### LINKED ISSUE #1175: Add badge for pre-commit.ci
By the way, I've gone ahead and enabled pre-commit.ci for this repo.

Here's a badge to integrate into README:
```rst
image:: https://results.pre-commit.ci/badge/github/aio-libs/aiobotocore/master.svg
   :target: https://results.pre-commit.ci/latest/github/aio-libs/aiobotocore/master
   :alt: pre-commit.ci status
```

_Originally posted by @webknjaz in https://github.com/aio-libs/aiobotocore/issues/1173#issuecomment-2305645667_

### PR DESCRIPTION
**Pull request by jakob-keller**
### Description of Change
Closes #1168, closes #1175

### Assumptions
Strings in this repo's YAML documents are frequently encapsulated in a pair of single or double quotes, which is unnecessary in many cases. Harmonizing those is not in scope of this PR.

### Checklist for All Submissions
* [ ] I have added change info to [CHANGES.rst](https://github.com/aio-libs/aiobotocore/blob/master/CHANGES.rst)
* [ ] If this is resolving an issue (needed so future developers can determine if change is still necessary and under what conditions) (can be provided via link to issue with these details):
  * [ ] Detailed description of issue
  * [ ] Alternative methods considered (if any)
  * [ ] How issue is being resolved
  * [ ] How issue can be reproduced
* [ ] If this is providing a new feature  (can be provided via link to issue with these details):
  * [ ] Detailed description of new feature
  * [ ] Why needed
  * [ ] Alternatives methods considered (if any)

### Checklist when updating botocore and/or aiohttp versions

* [ ] I have read and followed [CONTRIBUTING.rst](https://github.com/aio-libs/aiobotocore/blob/master/CONTRIBUTING.rst#how-to-upgrade-botocore)
* [ ] I have updated test_patches.py where/if appropriate (also check if no changes necessary)
* [ ] I have ensured that the awscli/boto3 versions match the updated botocore version

### CODE REVIEW

File: .github/FUNDING.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
+---
 # These are supported funding model platforms
 
+# yamllint disable rule:comments rule:line-length
```
Comment (webknjaz) [inline .github/FUNDING.yml:4]: You could instead just drop the comments and this wouldn't be necessary.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
+  - id: trailing-whitespace
+- repo: 'https://github.com/PyCQA/isort'
+  rev: 5.12.0
+  hooks:
+  - id: isort
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.4.8
+  hooks:
+  - id: ruff
+    args: [--fix]
```
Comment (webknjaz) [inline .pre-commit-config.yaml:17]: For the future: formatters should go before linters checking the same files. I also soft the checks by speed where possible.

**Comment by webknjaz** [comment]
By the way, I've gone ahead and enabled pre-commit.ci for this repo.

Here's a badge to integrate into README:
```rst
image:: https://results.pre-commit.ci/badge/github/aio-libs/aiobotocore/master.svg
   :target: https://results.pre-commit.ci/latest/github/aio-libs/aiobotocore/master
   :alt: pre-commit.ci status
```

File: .github/FUNDING.yml (hunk 1 above)
Comment (jakob-keller) [reply .github/FUNDING.yml:4]: That's true, but I don't consider the file meaningful to this project. I assume it is based on a GitHub template and the comment might still be useful to someone in the future.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (webknjaz) [reply .pre-commit-config.yaml:17]: Oh, I didn't realize that you inherit linters from them..

File: .github/FUNDING.yml (hunk 1 above)
Comment (webknjaz) [reply .github/FUNDING.yml:4]: It's just usually better not to have any ignores. Whenever you have an ignore, it should be accompanied with a code comment explaining why it's there and why can't it just be fixed.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (jakob-keller) [reply .pre-commit-config.yaml:17]: Yes, also several of the dev dependency pins, such as `pytest`.

File: .github/FUNDING.yml (hunk 1 above)
Comment (jakob-keller) [reply .github/FUNDING.yml]: I guess you're right! Comments are gone. Future someone should be able to look this up in git history or GitHub docs.

**Pull request merged**